homeassistant/components/traccar_server/coordinator.py

```diff
@@ -165,3 +202,57 @@ async def import_events(self, devices: list[DeviceModel]) -> None:
                     "attributes": event["attributes"],
                 },
             )
+
+    async def unsubscribe(self, *args) -> None:
+        """Unsubscribe from Traccar Server."""
+        if self._subscription is None:
+            return
+        self._should_log_subscription_error = False
+        self._subscription.cancel()
+        self._subscription = None
+
+    async def subscribe(self) -> None:
+        """Subscribe to events."""
+        if self._subscription is not None:
+            return
+
+        async def _subscriber():
+            try:
+                await self.client.subscribe(self.handle_subscription_data)
+            except TraccarException as ex:
+                if self._should_log_subscription_error:
+                    self._should_log_subscription_error = False
+                    LOGGER.error("Error while subscribing to Traccar: %s", ex)
+                # Retry after 10 seconds
+                await asyncio.sleep(10)
+                await _subscriber()
+
+        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self.unsubscribe)
+        self.config_entry.async_on_unload(self.unsubscribe)
+        self._subscription = asyncio.create_task(_subscriber())
```

Comment:
We could possibly use `hass.config_entries.async_create_background_task` here.

It will handle canceling the task on unload and shutdown.

Reply:
Cool, did not know of that one, will give it a try.